File: 7. Reverse Integer/reverse_number.swift
func reverse_number(num: Int, base: Int = 10) -> Int {
	var negative: Bool = num < 0;
	
	var number: Int = (num < 0) ? (num * -1) : (num * 1);
	
	var out: Int = 0;
	var i: Int = 1;
	var mod: Int;
	
	while number > 0{
		out *= i;
		mod = number % base;
		out += mod;
		number -= mod;
		number /= base;
		i = 10;
	}
	
	if (negative){
		out *= -1;
	}
	
	return (out);
}

print (reverse_number(num:-234567));

File: 7. Reverse Integer/reverse_number.c
#include <stdio.h>

int reverse_number(int number){
	int base = 10;
	int negative = (number < 0);
	
	number = (number < 0) ? (number * -1) : (number * 1);
	
	int nh = number;
	int out = 0;
	int i = 1;
	int mod;
	
	do {
		out *= i;
		mod = nh % base;
		out += mod;
		nh -= mod;
		nh /= base;
		i = 10;
	} while (nh > 0);
	
	if (negative){
		out *= -1;
	}
	
	return (out);
}

void main () {
	int number = -120;
	
	printf("%d\n\n\n", reverse_number(number));
}
